### DISCUSSION TITLE
API calling in Automation Action

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#5171] [Help] [answered]
Is there any way I can do POST API request in automation flow after the trigger is set . 
The flow is when there is data matching the set condition then there must be a POST API request that needs to be made

**Comment by Topener** [comment, marked as answer]
Hi @charan-dehaat, yes this is possible through the [REST data source](https://docs.budibase.com/docs/rest). 

- First, create a new REST data source.
- Next, create a new REST query using this setup:

<img width="996" alt="Screenshot 2022-03-30 at 10 16 43" src="https://user-images.githubusercontent.com/1898949/160784821-effa40d7-c817-4ec4-9c83-a83fb6fbd495.png">

- Then, in automation, use this setup

<img width="528" alt="Screenshot 2022-03-30 at 10 17 10" src="https://user-images.githubusercontent.com/1898949/160784901-a100e98d-1621-40f9-bb06-1e1bac66a6d9.png">

Let me know if you have issues setting this up! Any other parameter can also be configured with [bindings](https://docs.budibase.com/docs/rest-bindings)

**Comment by webserviceXXL** [comment]
Ahh, good one. Just had the same requirement and tried to solve this with the "JS scripting" action. But it seems that there are some limitations. I always get the error `XMLHttpRequest is not defined`. So I guess this isn't allowed in general. Are there any plans to allow request in the JS scripting, or is this on purpose?

**Reply by mjashanks** [reply]
This is on purpose. We have sandboxed the Javascript block for security reasons.  Any external requests should go via a dataconnector.

The purpose of the JS scripting block is to allow the manipulation of data from previous steps.
